airflow/operators/python.py

```diff
@@ -145,6 +145,69 @@ def execute_callable(self):
         return self.python_callable(*self.op_args, **self.op_kwargs)
 
 
+class PythonFunctionalOperator(PythonOperator):
+    """
+    Allows a workflow to "branch" or follow a path following the execution
+    of this task.
+
+    It derives the PythonOperator and expects a Python function that returns
+    a single task_id or list of task_ids to follow. The task_id(s) returned
+    should point to a task directly downstream from {self}. All other "branches"
+    or directly downstream tasks are marked with a state of ``skipped`` so that
+    these paths can't move forward. The ``skipped`` states are propagated
+    downstream to allow for the DAG state to fill up and the DAG run's state
+    to be inferred.
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        multiple_outputs: bool = False,
+        *args,
+        **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.multiple_outputs = multiple_outputs
+        
+
+    def __call__(self, *args, **kwargs):
+        self.op_args = args
+        self.op_kwargs = kwargs
+        # To add when we get XCom merged
+        # return XComArg(self)
+
+    def alias(self, task_id: str):
+        return PythonFunctionalOperator(
+            python_callable=self.python_callable, task_id=task_id
+        )
+
+    def execute(self, context: Dict):
+        return_value = self.execute_callable()
+        self.log.info("Done. Returned value was: %s", return_value)
+        if not self.multiple_outputs
+            return return_value
+        if isinstance(return_value, dict):
+            for key, value in return_value.items():
+                self.xcom_push(context, str(key), value)     
```

Comment:
[discussion] - the future problem is going to be around:  { date1: some_value}  with the date1:datetime,  (people like to have this kind of code for all kind of data mapping ).. and they will have to know that they should do `result['str_value_of_date']`.  This is just a problem, the solution is probably just in the documentation. We should not forget to describe that all keys of returned keys are going to be saved as str.

Reply:
We can also stringify key in `__getitem__` from XComArg. Thoughts?

Reply:
yes, this is a good idea.